redisinsight/ui/src/components/onboarding-tour/OnboardingTour.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useEffect, useState } from 'react'
+
+import {
+  EuiText,
+  EuiTourStep,
+  EuiButtonEmpty,
+  EuiButton, EuiButtonIcon,
+} from '@elastic/eui'
+import { useDispatch } from 'react-redux'
+import cx from 'classnames'
+
+import {
+  skipOnboarding,
+  setOnboardNextStep,
+  setOnboardPrevStep
+} from 'uiSrc/slices/app/features'
+import { Props as OnboardingWrapperProps } from './OnboardingTourWrapper'
+
+import styles from './styles.module.scss'
+
+export interface Props extends OnboardingWrapperProps {
+  isActive: boolean
+  currentStep: number
+  totalSteps: number
+}
+
+const OnboardingTour = (props: Props) => {
+  const {
+    options,
+    children,
+    anchorPosition = 'rightUp',
+    panelClassName,
+    anchorWrapperClassName,
+    isActive,
+    currentStep,
+    totalSteps,
+    preventPropagation,
+    fullSize
+  } = props
+  const { step, title, Inner } = options
+  const { content = '', onBack, onNext, onSkip } = Inner()
```

Comment:
it's may be dangerous. Butter change to `Inner?.()`

Reply:
changed, but in another way. With `Inner?.()` it will fail